### PR TITLE
Line.angle attribute 

### PR DESCRIPTION
**Pull request by ScriptLineStudios**
An angle attribute for Lines

### CODE REVIEW

File: src_c/line.c [hunk 1]
```diff
@@ -7,6 +7,9 @@
 #include <stddef.h>
 #include <math.h>
 
+#define PI 3.14159265358979323846264
```
Comment (Emc2356) [inline src_c/line.c:10]: wrap it in #ifndef PI as PI my be defined in other places also

File: src_c/line.c [hunk 2]
```diff
@@ -7,6 +7,9 @@
...
+#define RAD_TO_DEG(x) (x * 180 / PI)
```
Comment (Emc2356) [inline src_c/line.c:11]: again, add this in #ifndef

File: src_c/line.c [hunk 3]
```diff
@@ -916,11 +919,27 @@ pg_line_setb(pgLineObject *self, PyObject *value, void *closure)
...
+{
+    double dx = self->line.x2 - self->line.x1;
+    double dy = self->line.y2 - self->line.y1;
+
+    if (dx == 0.0) {
+        return (self->line.y2 > self->line.y1) ? PyFloat_FromDouble(-90.0)
+                                               : PyFloat_FromDouble(90.0);
+    }
+
+    double gradient = (dy / dx);
```
Comment (Emc2356) [inline src_c/line.c:933]: why parens

**Review by Emc2356 — COMMENTED** [review]
just this things and i think this is fine

File: src_c/line.c [hunk 4]
```diff
@@ -949,11 +956,26 @@ pg_line_setb(pgLineObject *self, PyObject *value, void *closure)
     return -1;
 }
 
+static PyObject *
+pg_line_getangle(pgLineObject *self, void *closure)
+{
+    double dx = self->line.x2 - self->line.x1;
+    double dy = self->line.y2 - self->line.y1;
```
Comment (itzpr3d4t0r) [inline src_c/line.c:963]: Not that it matters much, but you should move this `dy` calclation after the if statement.

**Pull request merged**